File: xjlib/src/main/java/me/xjliao/xjlib/BaseApp.java
/*
 * Copyright (c) 2017 xjliao.me created by xjliao
 * ProjectName: xjl
 * ModuleName: xjlib
 * FileName: BaseApp.java
 * ClassName: BaseApp
 * LastModified: 10/19/17 8:50 AM
 */

package me.xjliao.xjlib;

import android.app.Application;
import android.content.Context;
import android.support.multidex.MultiDex;

import com.squareup.otto.Bus;
import com.squareup.otto.ThreadEnforcer;
import com.squareup.picasso.Picasso;

/**
 * @author xjliao
 */
public class BaseApp extends Application {

    public static Context applicationContext;

    private static BaseApp instance;

    private static Bus EVENT_BUS;

    public static Context getAppContext() {
        return applicationContext;
    }

    public static Bus getEventBus() {
        return EVENT_BUS;
    }

    public static BaseApp getInstance() {
        return instance;
    }

    public void onCreate() {
        super.onCreate();
        applicationContext = this;
        instance = this;
        EVENT_BUS = new Bus(ThreadEnforcer.ANY);
        BaseApp.getEventBus().register(this);

        if (BuildConfig.DEBUG) {
            Picasso.with(this).setIndicatorsEnabled(true);
        }
    }

    public static void exit() {
        System.exit(0);
    }

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        MultiDex.install(this);
    }

}

File: xjlib/src/main/java/me/xjliao/xjlib/BaseApp.kt
/*
 * Copyright (c) 2017 xjliao.me created by xjliao
 * ProjectName: xjl
 * ModuleName: xjlib
 * FileName: BaseApp.java
 * ClassName: BaseApp
 * LastModified: 11/30/17 4:39 PM
 */

package me.xjliao.xjlib

import android.app.Application
import android.content.Context
import android.support.multidex.MultiDex

import com.squareup.otto.Bus
import com.squareup.otto.ThreadEnforcer
import com.squareup.picasso.Picasso

/**
 * @author xjliao
 */
open class BaseApp : Application() {

    override fun onCreate() {
        super.onCreate()
        appContext = this
        instance = this
        eventBus = Bus(ThreadEnforcer.ANY)
        BaseApp.eventBus!!.register(this)

        if (BuildConfig.DEBUG) {
            Picasso.with(this).setIndicatorsEnabled(true)
        }
    }

    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(base)
        MultiDex.install(this)
    }

    companion object {

        var appContext: Context? = null

        var instance: BaseApp? = null
            private set

        var eventBus: Bus? = null
            private set

        fun exit() {
            System.exit(0)
        }
    }

}